lib/active_record_encrypted_string/type.rb

```diff
@@ -28,6 +28,10 @@ def deserialize(value)
       v.present? ? encryptor.decrypt_and_verify(v) : v
     end
 
+    def changed_in_place?(raw_old_value, new_value)
+      deserialize(raw_old_value) != new_value
```

Comment:
I think we should take [same way](https://github.com/rails/rails/blob/7331881807fe07779c74eab332b0ba14419c5519/activemodel/lib/active_model/type/string.rb#L8-L12) as ActiveModel::Type::String.


```suggestion
      if new_value.is_a?(::String)
        deserialize(raw_old_value) != new_value
      end
```

Reply:
[Doc](https://api.rubyonrails.org/classes/ActiveModel/Type/Value.html#method-i-changed_in_place-3F) says `pass raw_old_value to Value#deserialize and compare it to new_value` .

Reply:
> > I did not write any tests because somehow, serialized is called twice within rspec example so its changing the original value :(
> 
> Could you tell me which code you mean??

The following test does not work in rspec because serialize in called twice within the class so its considered changed :(

We're live in production with my PR and I can assure you that the `changed_in_place` makes a difference but not in RSpec
```ruby
 context 'encrypt update' do
      let(:value) { :encrypted }
      let(:dummy_klass) do
        Class.new(ActiveRecord::Base) do
          self.table_name = 'dummys'

          attribute :encrypted, :encrypted_string
        end
      end

      it 'should not update encrypted when not changed' do
        instance.save!
        p instance.save.explain
        expect { instance.save! }.not_to change { instance.read_attribute_before_type_cast(:encrypted) }
      end
    end
```

Reply:
> The following test does not work in rspec because serialize in called twice within the class so its considered changed :(

Confirmed what you say. I think activerecord calls serialize method twice when activerecord executes `instance.save!` at the first time not my gem process. Also I realized the value of encrypted column is saved to database as first encryption value. It's mean the following test is works.

```rb
context 'encrypt update' do
  let(:value) { :encrypted }
  let(:dummy_klass) do
    Class.new(ActiveRecord::Base) do
      self.table_name = 'dummys'

      attribute :encrypted, :encrypted_string
    end
  end

  it 'should not update encrypted when not changed' do
    instance.save!
    expect { instance.save! }.not_to change { instance.reload.read_attribute_before_type_cast(:encrypted) }
  end
end
```

Reply:
I know it's know your gem the cause. I just tried to make an rspec, but ain't working :( Anyway, the `changed_in_place` is enough :)

Reply:
> I know it's know your gem the cause

Let me know which code is. Because I don't want to accept any changes without test codes.

Reply:
Also, I'm sorry for my late reply.